### PR TITLE
gitoxide: init at 0.3.0

### LINKED ISSUE #95317: gitoxide
**Project description**

An idiomatic, modern, lean, fast, safe & pure rust implementation of git 

**Metadata**

* homepage URL: https://github.com/Byron/gitoxide
* source URL: https://github.com/Byron/gitoxide
* license: mit
* platforms: all

### PR DESCRIPTION
**Pull request by syberant**
###### Motivation for this change
This PR is intended to fix #95317.

###### Things I'm not too sure about
The project is pretty young at the moment, I've only had 1 PR merged into nixpkgs so far so I'm not entirely sure what is supposed to go in here and what gets rejected. Some feedback would be appreciated.

I got the `meta.description` from the projects [crates.io page](https://crates.io/crates/gitoxide) but there might be a better one.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`), **but only the help subcommand**.
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pstn** [comment]
If you want to, you can add a `Fixes:` line to your commit. That will to github closing that issue when the commit is merged.
It should look like [this recent example](https://github.com/NixOS/nixpkgs/commit/4a966b35c11be14332655452991904c5c9a4aa42).

**Review by TomSmeets — APPROVED** [review]
Looks good. 

But, as you said the project appears to be very young. As far as I know currently only `gix init` is implemented.

**Comment by syberant** [comment]
> If you want to, you can add a `Fixes:` line to your commit. That will to github closing that issue when the commit is merged.

@pstn, I think `fix #95317` in this PR already does the job for GitHub, is it considered cleaner to do it in the commit message instead?

**Comment by alyssais** [comment]
> > If you want to, you can add a `Fixes:` line to your commit. That will to github closing that issue when the commit is merged.
> 
> @pstn, I think `fix #95317` in this PR already does the job for GitHub, is it considered cleaner to do it in the commit message instead?

It’s helpful for people going through the commit history. It’s also nice to include the full issue URL (which GitHub will still recognise and even abbreviate in the UI!) so it’s clickable and clearer if we ever move to a different system.

**Comment by syberant** [comment]
Thanks, I included it into the commit message.

File: pkgs/applications/version-management/gitoxide/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "Byron";
+    repo = "gitoxide";
+    rev = "v${version}";
+    sha256 = "0xpic9jx7nrxi5d8lqch2vxpvipx994d717c4n0kgr3ipyij1347";
+  };
+
+  verifyCargoDeps = true;
```
Comment (zowoq) [inline pkgs/applications/version-management/gitoxide/default.nix:14]: `verifyCargoDeps` was removed from `buildRustPackage`.
```suggestion
```

Comment (syberant) [reply pkgs/applications/version-management/gitoxide/default.nix]: Thanks, I completely missed that development.

File: pkgs/applications/version-management/gitoxide/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  cargoSha256 = "104lyfni75h1i30s2jlzf66sp1czfd9ywqz78kj4i7lfdf6fc4x9";
+
+  meta = with lib; {
+    description =
+      "A command-line application for interacting with git repositories";
+    homepage = "https://github.com/Byron/gitoxide";
+    # NOTE: the master branch is dual-licensed with APACHE but v0.3.0 is only MIT
+    license = licenses.mit;
+    maintainers = [ maintainers.syberant ];
+    platforms = platforms.all;
```
Comment (sternenseemann) [inline pkgs/applications/version-management/gitoxide/default.nix:23]: You can omit this, as it's set by `buildRustPackage` to the appropriate setting for this case:

https://github.com/NixOS/nixpkgs/blob/6d0a85fe528e020eb7d8eb78d33ee87055c01c74/pkgs/build-support/rust/default.nix#L246

**Review by sternenseemann — CHANGES_REQUESTED** [review]
Otherwise LGTM, builds fine for me on NixOS, tested `gix init` seems good.

File: pkgs/applications/version-management/gitoxide/default.nix (hunk 2 above)
Comment (syberant) [reply pkgs/applications/version-management/gitoxide/default.nix]: Ah, good to know.

**Comment by sternenseemann** [comment]
Probably good idea to squash the 4 commits into one or two, otherwise lgtm.

**Comment by syberant** [comment]
I've squashed the three commits about gitoxide together but left the changes to maintainer-list as their own commit.

**Comment by alyssais** [comment]
@GrahamcOfBorg build gitoxide

**Comment by syberant** [comment]
It seems to be stuck at the darwin check, after inspecting https://monitoring.nix.ci/d/000000002/ofborg?orgId=1&refresh=10s I think that is because there is currently no builder for darwin.

**Comment by turion** [comment]
Awesome, thanks @syberant!

**Comment by syberant** [comment]
No problem @turion, I saw it on reddit and thought it an interesting project. Your issue was the final push.

**Pull request merged**